Add support for fetching a new persistent license when one already exists

jaredmcneill opened the pull request:
Add a new API method `updatePersistent`, which is like `createPersistent` except it skips the test for a persistent already existing.

Our application requests time restricted licenses for offline playback and periodically refreshes licenses in the background. Without this change, we need to do a `removePersistent` before `createPersistent`, which creates a short window where playback is not possible while the license is being updated.

obecny commented:
hey @jaredmcneill thx for contribution but I think that adding an extra api might be a bit confusing and might make the `createPersistent` api redundant.
There is also one thing that is missing here which is the removing the previous persistent session. I can simply assume that if you create a new one the previous one should be removed otherwise it will stay there forever and information about its id will be lost too.

So my proposition to this would be the following:

1. Extend the `createPersistent` by simply adding the 3rd param `forced` a boolean type, this would be the optional and by default it will be `false`.
2. Once someone will call `createPersistent('myId', 'myConfig', true)` it will follow the same flow except raising the error if persistent session exist, but it will also remove the previous persistent session if such existed the flow would look like this inside the `createPersistent` function:
a) if forced then save in variable the information about previous persistent session id for example `existingPersistentSessionId` and don't raise the error, otherwise if user didn't add forced it should behave as it is now
b) create a new persistent session
c) save the new persistent session id
d) remove the `existingPersistentSessionId` if `forced` was true and previous persistent session existed
e) and finally resolve the the promise `createPersistent`

jaredmcneill commented:
Thanks for the review @obecny -- adding a parameter to `createPersistent` is fine with me. I'll make changes based on your recommendations and submit an updated patch.

jaredmcneill commented:
Hi @obecny - I've removed `updatePersistent` and added the `forced` flag to createPersistent.

Can you explain what you mean by leaving old persistent sessions around forever? I tried adding a `removePersistent` call after `savePersistent` to remove the old session but it appears to have clobbered the new one despite passing the old ID.

obecny commented:
`removePersistent` and  `savePersistent` is a part of downstream api so this is correct it removes the current persistent session. You have to remove the old one and I think you should use api from persistent plugin, that is what I assume you have mixed up.

jaredmcneill commented:
Thank you @obecny -- I've updated the patch with your suggestions.

obecny commented:
@jaredmcneill looks fine, just one thing for easier testing the api, can you add a button for example "Create Persistent Session - forced" so that it will be available for testing ? the same way it works for other stuff and it will be a complete then. :) thx

jaredmcneill commented:
Done!
